modPrecompiled/src/org/aion/precompiled/ContractFactory.java

```diff
@@ -103,6 +135,25 @@ public static boolean isPrecompiledContract(Address address) {
      * @return the contract address.
      */
     public static Address getTotalCurrencyContractAddress() {
-        return Address.wrap(TOTAL_CURRENCY);
+        return Address.wrap(ADDR_TOTAL_CURRENCY);
     }
+
+    /**
+     * Returns the address of the EdVerifyContract contract.
+     *
+     * @return the contract address
+     */
+    public static Address getEdVerifyContractAddress() {
+        return Address.wrap(ADDR_ED_VERIFY);
+    }
+
+    /**
+     * Returns the address of the TxHash contract.
+     *
+     * @return the contract address
+     */
+    public static Address getTxHashContractAddress() {
+        return Address.wrap(ADDR_TX_HASH);
+    }
+
```

Comment:
shouldn't a getter for the blacke2b address be added here as well?

Reply:
added